### PR TITLE
add ci and golangci-lint

### PR DESCRIPTION
**Pull request by pperiyasamy**
Signed-off-by: Periyasamy Palanisamy <periyasamy.palanisamy@est.tech>

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+  max-issues-per-linter: 0
+  max-same-issues: 0
+  exclude-rules:
+    # We really *do* want to pass a pointer to an interface in these cases.  See comments in file for New{Server,Client}
+    # function for why
+    - path: .*\.gen\.go
+      linters:
+        - dupl
+        - golint
+    - path: .*registry.*.go
```
Comment (Bolodya1997) [inline .golangci.yml:176]: I am not actually sure that we need lines down from here in `sdk-ovs`, because this excludes are specific for `sdk`.

Comment (pperiyasamy) [reply .golangci.yml]: yes @Bolodya1997 , removed it now :)

File: .golangci.yml [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    - typecheck
+    - unconvert
+    - unparam
+    # - unused
+    - varcheck
+    - whitespace
+issues:
+  exclude-use-default: false
+  max-issues-per-linter: 0
+  max-same-issues: 0
```
Comment (Bolodya1997) [inline .golangci.yml:168]: Please add here a newline at the end - it will be an error for `yamlint` when it will be enabled.

Comment (pperiyasamy) [reply .golangci.yml]: done.

**Review by Bolodya1997 — APPROVED** [review]

**Review by denis-tingaikin — APPROVED** [review]
Thanks, @Bolodya1997 , @pperiyasamy !

**Pull request merged**